BrainPortal/app/controllers/users_controller.rb

```diff
@@ -141,12 +141,13 @@ def create #:nodoc:
       flash[:notice] = "User successfully created."
 
       # Find signup record matching login name, and log creation and transfer some info.
-      if signup = Signup.where(:login => @user.login, :approved_at => nil, :approved_by => nil).first
+      if signup = Signup.where(:id => params[:signup_id]).first
```

Comment:
Why remove the safety checks for the record not having been approved before? The way it is set up it if there is a mixup in the signup_id, the wrong record will be adjusted.